### ISSUE TITLE
Subscribe request doesn't work properly

### ISSUE THREAD

**Issue by solartune** [slact/nchan#450]
Hello @slact !
I have a problem with using nchan.
When I publish messages with `nchan_subscribe_request` directive and subscribe to a channel then a subscriber can see only one (latest) message. If there are several different messages the subscriber will see only latest one but duplicated. 

For example. I sent 4 messages (message1 - message4). When I opened the page I see 2 duplicates of the last one. And there are errors in the console log. When I send messages with open connection it's ok. I see each message (message5-message7). With no errors.
![image](https://user-images.githubusercontent.com/7710618/38552139-754c7d02-3cbb-11e8-8a03-ef8bfa3b3f17.png)

However when I disable `nchan_subscribe_request` directive it works fine.
![image](https://user-images.githubusercontent.com/7710618/38552461-77ef43b8-3cbc-11e8-9170-001e405b9159.png)

And this happens only with `nchan_subscribe_request`.  With `nchan_UNsubscribe_request` it works fine.

Nginx version: 1.13.10
Nchan version: 1.1.14

In the config bellow I subscribe directly to a channel (there are different behaviors between direct and accel redirect subscribing, but I'll describe it in another one issue)

My config:
```nginx
nchan_redis_url "redis://redis:6379";
...
server {
    ...
    location ~ /pub/user/(\w+)$ {
        nchan_publisher;
        nchan_channel_id user_$1;
        nchan_message_buffer_length 10;
        nchan_use_redis on;
    }

    location ~ /pub/global$ {
        nchan_publisher;
        nchan_channel_id global;
        nchan_message_buffer_length 10;
        nchan_use_redis on;
    }

    location /sub_upstream {
        proxy_pass http://localxmr-backend:8000/channels/subscriber_x_accel_redirect/;
        proxy_set_header X-Forwarded-For $remote_addr;
     }

    # Subscriber endpoint used by clients.
    location ~ /sub/user/(\w+)$ {
         # this location only accessible for internal nginx redirects
         # TODO a temporary allow to subscribe directly
         # internal;
         nchan_subscriber websocket;
         # clients subscribe to both user-specific and global messages.
         nchan_channel_id user_$1 global;
         nchan_use_redis on;
         nchan_unsubscribe_request /upstream/unsub;
         # TODO doesn't work properly
         # nchan_subscribe_request /upstream/sub;
    }
    location = /upstream/unsub {
         proxy_pass http://localxmr-backend:8000/channels/unsub/;
         proxy_ignore_client_abort on;
         proxy_set_header X-Subscriber-Type $nchan_subscriber_type;
         proxy_set_header X-Channel-Id $nchan_channel_id;
         proxy_set_header X-Original-URI $request_uri;
    }

    location = /upstream/sub {
         proxy_pass http://localxmr-backend:8000/channels/sub/;
         proxy_set_header X-Subscriber-Type $nchan_subscriber_type;
         proxy_set_header X-Message-Id $nchan_message_id;
         proxy_set_header X-Channel-Id $nchan_channel_id;
         proxy_set_header X-Original-URI $request_uri;
    }
```

**Comment by slact** [comment]
I'm unable to reproduce this. Does this happen with and without Redis? I suspect this is related to the fix I referenced in #451 . Please check if this happens without Redis, and with a build from master.

**Comment by solartune** [comment]
>Please check if this happens without Redis, and with a build from master.

It happens with and without Redis even with the master version

**Comment by solartune** [comment]
I enabled subscribe/unsubscribe requests using wscat:
Before:
![image](https://user-images.githubusercontent.com/7710618/42509201-73b0e118-844b-11e8-94e1-0376a63ec43c.png)

After:
![image](https://user-images.githubusercontent.com/7710618/42509208-774b8be8-844b-11e8-820c-48abcd5dc76c.png)

As you can see all messages began to look like the last

**Comment by tihoho** [comment]
@solartune Hey! tell me how you implemented tracking online / offline user states?

**Comment by solartune** [comment]
> @solartune Hey! tell me how you implemented tracking online / offline user states?

If I remember correctly, we abandoned tracking the online state of the users with Nchan eventually as it was quite unreliable at the time but if you take a look at the config above:
1. We used `nchan_unsubscribe_request` and `nchan_subscribe_request` that were pointing at the nginx locations.
2. Both locations had proxy_pass pointing at appropriate endpoints on the backend side.
3. When the endpoint would be called we would change the user's state.

That's from the top of my head as it was done 3 years ago I don't work on the project anymore. So I don't really remember details :D 

Hope that helps.

**Comment by tihoho** [comment]
@solartune yes I understand it all. the endpoint accepts headers with the channel ID. I do not understand how exactly to track a specific user if he does not have an id. in other words, the process in point 3 is interesting.

Thanks for feedback.

**Comment by solartune** [comment]
> @solartune yes I understand it all. the endpoint accepts headers with the channel ID. I do not understand how exactly to track a specific user if he does not have an id. in other words, the process in point 3 is interesting.
> 
> Thanks for feedback.

There would be an auth token in the user's cookies. So with that token, we were able to identify the user.
